library/src/text/shaping.rs

```diff
@@ -679,6 +731,54 @@ fn track_and_space(ctx: &mut ShapingContext) {
     }
 }
 
+pub fn is_gb_style(styles: StyleChain) -> bool {
+    let lang = TextElem::lang_in(styles);
+    let region = TextElem::region_in(styles);
+
+    // Most CJK variants, including zh-CN, ja-JP, zh-SG, zh-MY use GB-style punctuation,
+    // while zh-HK and zh-TW use alternative style. We default to use GB-style.
+    !(lang == Lang::CHINESE
+        && matches!(region.as_ref().map(Region::as_str), Some("TW" | "HK")))
+}
+
+/// Calculate stretchability and shrinkability of each glyph,
+/// and CJK punctuation adjustments according to Chinese Layout Requirements.
+fn calculate_adjustability(ctx: &mut ShapingContext) {
+    let gb_style = is_gb_style(ctx.styles);
+
+    let mut glyphs = ctx.glyphs.iter_mut().peekable();
+    while let Some(glyph) = glyphs.next() {
+        glyph.adjustability = glyph.base_adjustability(gb_style);
+
+        if glyph.is_cjk_adjustable() {
+            // Only GB style needs further adjustment.
+            if !gb_style {
+                continue;
+            }
```

Comment:
```suggestion
        // Only GB style needs further adjustment.
        if gb_style && glyph.is_cjk_adjustable() {
            continue;
        }
```
and then dedent the rest